### ISSUE TITLE
Array destructuring issue

### ISSUE THREAD

**Issue by zamb3zi** [babel/babili#607]
This is very similar to #151 and is occurring on the current master version.

### Input ###

```js
function foo(arg) {
    const {
        x: [x],
        y: [y],
    } = arg;
    return x;
}
```

### Output ###
```js
function foo(a){return b}
```
[REPL](https://babeljs.io/repl/#?babili=true&evaluate=true&lineWrap=false&presets=es2016&targets=&browsers=&builtIns=false&debug=false&code_lz=GYVwdgxgLglg9mABMOcAUBDATgcwJSIDeAUImYhAgM5RGnkMAeAXIgNqMC6ANPQ2QE9WbATz6IAvogC8ibDgDc4rAFMoILEkZKJxIA)

Compared with the following correct behaviour:
### Input ###

```js
function foo(arg) {
    const {
        x: [x],
        y: [y],
    } = arg;
    return x + y;
}

```

### Output ###
```js
function foo(a){const{x:[b],y:[c]}=a;return b+c}
```
[REPL](https://babeljs.io/repl/#?babili=true&evaluate=true&lineWrap=false&presets=es2016&targets=&browsers=&builtIns=false&debug=false&code_lz=GYVwdgxgLglg9mABMOcAUBDATgcwJSIDeAUImYhAgM5RGnkMAeAXIgNqMC6ANPQ2QE9WbATz6IAvogC8ibDgDc4rAFMoILEkaIA1IgFKJxIA)

**Comment by boopathi** [comment]
Interesting. Thanks. Looks like a bug in DCE. Will try this.

**Comment by boopathi** [comment]
Fixed by #622

**Comment by zamb3zi** [comment]
Confirmed, my test cases are now working.
